server/svix-server/src/v1/endpoints/application.rs

```diff
@@ -148,18 +151,26 @@ async fn get_application(
     Path(app_id): Path<ApplicationIdOrUid>,
     permissions: Permissions,
 ) -> Result<Json<ApplicationOut>> {
-    let app = application::Entity::secure_find_by_id_or_uid(permissions.org_id, app_id)
-        .one(db)
-        .await?
-        .ok_or_else(|| HttpError::not_found(None, None))?;
+    // If someone authorized for only applications calls this endpoint, it will only return their
+    // application regardless of input in the path
```

Comment:
You should return the ID in the path (so change this back). The case where the ID doesn't match should be handled by the permission check as I mentioned in security.rs.